core/conversion/converters/impl/shuffle.cpp

```diff
@@ -67,33 +67,33 @@ static auto shuffle_registrations TORCHTRT_UNUSED =
         .pattern(
             {"aten::reshape(Tensor self, int[] shape) -> (Tensor)",
              [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
-               auto in = args[0].ITensorOrFreeze(ctx);
-               auto in_shape = util::toVec(in->getDimensions());
-               std::vector<int64_t> new_shape;
-               if (ctx->input_is_dynamic) {
-                 new_shape = util::toVec(args[1].unwrapToIntList().vec());
-                 int nbDynamicDims = 0;
-                 for (size_t i = 0; i < new_shape.size(); i++) {
-                   if (in_shape[i] == -1)
-                     nbDynamicDims++;
-                 }
-                 if (nbDynamicDims > 1) {
-                   TORCHTRT_THROW_ERROR(
-                       "Resize is currently not supported when target shape contains more than one dynamic dimension");
-                 }
-               } else {
-                 new_shape = torch::reshape(torch::rand(in_shape), args[1].unwrapToIntList().vec()).sizes().vec();
-               }
-
-               auto shuffle = ctx->net->addShuffle(*in);
-               TORCHTRT_CHECK(shuffle, "Unable to create shuffle layer from node: " << *n);
-               shuffle->setReshapeDimensions(util::toDims(new_shape));
-               shuffle->setName(util::node_info(n).c_str());
-
-               auto out_tensor = ctx->AssociateValueAndTensor(n->outputs()[0], shuffle->getOutput(0));
-               LOG_DEBUG("Output tensor shape: " << out_tensor->getDimensions());
-
-               return true;
+                auto in = args[0].ITensorOrFreeze(ctx);
+                auto in_shape = util::toVec(in->getDimensions());
+                std::vector<int64_t> new_shape;
+                nvinfer1::ITensor* shape_tensor;
+                if (ctx->input_is_dynamic) {
+                  auto new_shape = args[1].unwrapToITensorList();
```

Comment:
Should this check whether args[1] is an ITensorList? If we have some intermediate math on the shape input that is handled by evaluators could we end up in a case where input_is_dynamic but args[1] is an IntList input?

Reply:
Yeah WIP. Currently, at multiple places, we unwrap without detecting the types. Investigating.

Reply:
Added type checking now.